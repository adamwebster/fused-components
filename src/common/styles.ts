import styled from "styled-components";


export const DarkModeWrapper = styled.div`
  width: 100%;
  padding:30px 20px 30px 20px;
  background-color: #001331;
  box-sizing: border-box;
  border-radius: 5px;
  box-shadow: rgba(0,0,0,0.10) 0 1px 3px 0;
  margin: 25px 0 40px;
`
