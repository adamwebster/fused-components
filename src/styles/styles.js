export const color = {
    primary: '#4799ff',
    secondary: '#ff6347',
    tertiary: '#DDDDDD',
    danger: "#ffbebe",
    warning: "#ffe3c7",
    info: "#bcd7ff",
    success: "#c0dcb1",
    red: "#dd2c2e",
    yellow: "#ffc135",
    blue: "#1c91dc",
    green: "#79ca4c",
}