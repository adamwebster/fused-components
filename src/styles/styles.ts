import { lighten } from 'polished';

export const color = {
  primary: '#006CF0',
  primaryButton: '#C2DDFF',
  focus: '#4799ff',
  secondary: '#ff6347',
  tertiary: '#DDDDDD',
  danger: '#dd2c2e',
  dangerButton: '#F5B2B2',
  warning: '#eba300',
  warningButton: '#FFE7AD',
  info: '#2f83cf',
  infoButton: '#C0DAF2',
  success: '#00b76b',
  successButton: '#DBFEB5',
  red: '#dd2c2e',
  yellow: '#eba300',
  blue: '#205A8E',
  green: '#00b76b',
  highlight: lighten(0.3, '#4799ff'),
  // Monochrome
  lightest: '#FFFFFF',
  lighter: '#F8F8F8',
  light: '#F3F3F3',
  mediumlight: '#EEEEEE',
  medium: '#DDDDDD',
  mediumdark: '#999999',
  dark: '#666666',
  darker: '#444444',
  darkest: '#333333',
  overlay: 'rgba(0,0,0,.5)',
  border: 'rgba(0,0,0,.1)',
  darkModeBG: '#102A43',
  darkModeDarkest: '#102A43',
  darkModeDarker: '#243b53',
  darkModeDark: '#334e68',
  darkModeMediumDark: '#486581',
  darkModeMedium: '#627d98',
  darkModeMediumLight: '#829ab1',
  darkModeLight: '#9fb3c8',
  darkModeLighter: '#bcccdc',
  darkModeLightest: '#d9e2ec',
  darkModeBright: '#f0f4f8',
};
