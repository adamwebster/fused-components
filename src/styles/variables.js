const variables = {
  
  accentColor: '#4799ff',
  borderRadius: '5px'
 }
 
 export default variables;