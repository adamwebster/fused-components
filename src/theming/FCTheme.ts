import React from 'react';

export interface Theme {
    theme?: string,
    children?: any,
}

export const FCTheme = React.createContext<Theme | null>(null);

export const FCThemeProvider = FCTheme.Provider;

export const FCThemeConsumer = FCTheme.Consumer;