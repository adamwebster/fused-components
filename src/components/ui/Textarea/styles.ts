import styled from 'styled-components';
import { color } from '../../../styles/styles';
import { darken, lighten } from 'polished';

export const StyledTextarea = styled.textarea`
    border-radius: 5px;
    border: solid 1px ${props => props.theme === 'dark' ? darken(0.3, color.focus) : color.border};
    background-color: ${props => props.theme === 'dark' ? color.darkModeBG : '#fff'};

`
