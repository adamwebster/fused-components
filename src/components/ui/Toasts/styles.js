import styled from "styled-components";
import { Alert } from "../Alert/";
import { color } from "../../../styles/styles";

export const ToastContainer = styled.div`
  position: fixed;
  width: 500px;
  ${props => {
    let position = "";

    switch (props.position) {
      case "top":
        position = `
            top:20px;
            left: 50%;
            transform: translateX(-50%);
            `;
        break;
      case "bottom":
        position = `
             bottom:20px;
            left: 50%;
            transform: translateX(-50%);
            `;
        break;

      case "bottom-right":
        position = `
        bottom:20px;
        right:20px;
            `;
        break;
      case "top-right":
      default:
        position = `
        top:20px;
        right:20px;
            `;
    }
    return [position];
  }}

  z-index:99;
`;

export const StyledToast = styled(Alert)`
  margin-bottom: 10px;
  transition: all;
  position: relative;
  animation: ${props =>
    !props.removing
      ? "fadeinToast 0.5s ease-in-out"
      : "fadeoutToast 0.5s ease-in-out"};
  @keyframes fadeinToast {
    0% {
      transform: scale(0);
      opacity: 0;
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  @keyframes fadeoutToast {
    0% {
      transform: scale(1);
      opacity: 1;
    }
    100% {
      opacity: 0;
      transform: scale(0);
    }
  }
`;

export const LoadingBar = styled.div`
  position: relative;
  height: 5px;
  transition: all;
  margin-top: 10px;
  background-color: ${color.medium};
  width: ${props => props.timer}%;
`;

export const CloseButton = styled.button`
  width: 12px;
  box-sizing: border-box;
  padding: 0;
  border: none;
  color: ${color.mediumdark};
  background-color: transparent;
  position: absolute;
  top: 15px;
  right: 15px;
  cursor: pointer;
  &:hover {
    color: ${color.dark};
  }
  svg {
    color: inherit;
  }
`;
