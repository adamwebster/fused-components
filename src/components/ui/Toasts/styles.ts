import styled from 'styled-components';
import { Alert } from '../Alert';
import { color } from '../../../styles/styles';
import { lighten, darken } from 'polished';
import { fcStyles } from '../../../common/types';

interface ToastContainerInterface {
  position?: 'top' | 'bottom' | 'top-right' | 'bottom-right';
}
export const ToastContainer = styled.div<ToastContainerInterface>`
  position: fixed;
  ${(props): string[] => {
    let position = '';

    switch (props.position) {
      case 'top':
        position = `
            top:20px;
            left: 50%;
            transform: translateX(-50%);
            width: 500px;
            @media (max-width: 500px) {
              width: calc(100% - 20px);
            }
            `;
        break;
      case 'bottom':
        position = `
             bottom:20px;
            left: 50%;
            transform: translateX(-50%);
            width: 500px;
            @media (max-width: 500px) {
              width: calc(100% - 20px);
            }
            `;
        break;

      case 'bottom-right':
        position = `
        bottom:20px;
        right:20px;
        width: 300px;
            `;
        break;
      case 'top-right':
      default:
        position = `
        top:20px;
        right:20px;
        width: 300px;

            `;
    }
    return [position];
  }}

  z-index:99;
`;

interface ST extends React.HTMLProps<HTMLElement> {
  removing: boolean;
  timer: string;
  theme?: unknown;
}
export const StyledToast = styled(Alert)<ST>`
  margin-bottom: 10px;
  transition: all;
  position: relative;
  border-radius: 5px;
  animation: ${(props): string => (!props.removing ? 'fadeinToast 0.5s ease-in-out' : 'fadeoutToast 0.5s ease-in-out')};
  @keyframes fadeinToast {
    0% {
      transform: scale(0);
      opacity: 0;
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  @keyframes fadeoutToast {
    0% {
      transform: scale(1);
      opacity: 1;
    }
    100% {
      opacity: 0;
      transform: scale(0);
    }
  }
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
`;

interface LB {
  timer: string;
  theme?: unknown;
  fcStyle?: fcStyles;
}
export const LoadingBar = styled.div<LB>`
  position: relative;
  height: 5px;
  transition: all;
  margin-top: 10px;
  background-color: ${(props): string => {
    switch (props.fcStyle) {
      case 'danger':
        return darken(0.1, color.danger);
      case 'warning':
        return lighten(0.2, color.warning);
      case 'info':
        return lighten(0.2, color.info);
      case 'success':
        return lighten(0.1, color.success);
      default:
        return props.theme === 'dark' ? color.medium : color.light;
    }
  }};
  width: ${(props): string => props.timer}%;
`;

export const CloseButton = styled.button`
  width: 12px;
  box-sizing: border-box;
  padding: 0;
  border: none;
  color: ${color.medium};
  background-color: transparent;
  position: absolute;
  top: 15px;
  right: 15px;
  cursor: pointer;
  &:hover {
    color: ${color.light};
  }
  svg {
    color: inherit;
  }
`;
