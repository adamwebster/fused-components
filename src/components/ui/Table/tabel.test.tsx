import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import { Table } from './index';
import { color } from '../../../styles/styles';
import { FCThemeProvider } from '../../../theming/FCTheme';

afterEach(cleanup);

describe('Table Tests', () => {
  test('Renders the Table component', () => {
    const { getByText } = render(
      <Table>
        <Table.Header>
          <Table.Row>
            <Table.Cell>First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByText('Clark')).toBeInTheDocument();
    expect(getByText('First Name')).toBeInTheDocument();
  });

  test('Renders correct styles in dark mode', () => {
    const { container } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Table>
          <Table.Header>
            <Table.Row>
              <Table.Cell>First Name</Table.Cell>
              <Table.Cell>Last Name</Table.Cell>
              <Table.Cell>Job</Table.Cell>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            <Table.Row>
              <Table.Cell>Clark</Table.Cell>
              <Table.Cell>Kent</Table.Cell>
              <Table.Cell>Superman</Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>
      </FCThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Setting  the table header to have no border removes the border', () => {
    const { getByText } = render(
      <Table data-testid="table">
        <Table.Header showBottomBorder={false}>
          <Table.Row>
            <Table.Cell>First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByText('First Name').parentNode?.parentNode).toHaveStyleRule('border-bottom', undefined);
  });

  test('Setting table to have a padding changes the padding in the styles', () => {
    const { getByTestId } = render(
      <Table padding="10px" data-testid="table">
        <Table.Header showBottomBorder={false}>
          <Table.Row>
            <Table.Cell data-testid="table-cell">First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByTestId('table-cell')).toHaveStyleRule('padding', '10px');
  });

  test('Setting table to have zebra striping adds the zebra stripe color', () => {
    const { getByTestId } = render(
      <Table padding="10px" zebraStripeColor="tomato" zebraStriping data-testid="table">
        <Table.Header showBottomBorder={false}>
          <Table.Row>
            <Table.Cell data-testid="table-cell">First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
          <Table.Row data-testid="alt-row">
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByTestId('alt-row')).toHaveStyleRule('background-color', 'tomato', {
      modifier: ':nth-child(even)',
    });
  });
  test('First column freezes', () => {
    const { getByTestId } = render(
      <Table freezeFirstColumn padding="10px" data-testid="table">
        <Table.Header data-testid="table-header" showBottomBorder={false}>
          <Table.Row>
            <Table.Cell data-testid="table-cell">First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByTestId('table-cell')).toHaveStyleRule('position', 'absolute', { modifier: ':first-child' });
  });
  test('Highlight color shows on hover', () => {
    const { getByTestId } = render(
      <Table highlightOnHover data-testid="table">
        <Table.Header data-testid="table-header" showBottomBorder={false}>
          <Table.Row>
            <Table.Cell data-testid="table-cell">First Name</Table.Cell>
            <Table.Cell>Last Name</Table.Cell>
            <Table.Cell>Job</Table.Cell>
          </Table.Row>
        </Table.Header>
        <Table.Body data-testid="table-body">
          <Table.Row data-testid="table-row">
            <Table.Cell>Clark</Table.Cell>
            <Table.Cell>Kent</Table.Cell>
            <Table.Cell>Superman</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>,
    );
    expect(getByTestId('table-body')).toHaveStyleRule('background-color', color.highlight, { modifier: 'tr:hover' });
  });
  test('Highlight color shows on hover | Dark mode', () => {
    const { getByTestId } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Table highlightOnHover data-testid="table">
          <Table.Header data-testid="table-header" showBottomBorder={false}>
            <Table.Row>
              <Table.Cell data-testid="table-cell">First Name</Table.Cell>
              <Table.Cell>Last Name</Table.Cell>
              <Table.Cell>Job</Table.Cell>
            </Table.Row>
          </Table.Header>
          <Table.Body data-testid="table-body">
            <Table.Row data-testid="table-row">
              <Table.Cell>Clark</Table.Cell>
              <Table.Cell>Kent</Table.Cell>
              <Table.Cell>Superman</Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>
      </FCThemeProvider>,
    );
    expect(getByTestId('table-body')).toHaveStyleRule('background-color', color.darkModeMedium, {
      modifier: 'tr:hover',
    });
  });
});
