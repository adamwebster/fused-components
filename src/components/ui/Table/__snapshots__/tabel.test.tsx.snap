// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Tests Renders correct styles in dark mode 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  overflow: auto;
  color: #DDDDDD;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  font-weight: bold;
  border-bottom: solid 1px #999999;
  border-collapse: collapse;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  padding: 5px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

<div>
  <table
    class="c0"
  >
    <thead
      class="c1"
    >
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          First Name
        </td>
        <td
          class="c3"
        >
          Last Name
        </td>
        <td
          class="c3"
        >
          Job
        </td>
      </tr>
    </thead>
    <tbody
      class="c4"
    >
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          Clark
        </td>
        <td
          class="c3"
        >
          Kent
        </td>
        <td
          class="c3"
        >
          Superman
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
